### PR TITLE
Adds Riot Shield Bashing with Combat Knives

### PR DESCRIPTION
**Pull request by Bjamcham**
## What Does This PR Do
Adds combat knives alongside batons to the items that can bash riot shields for the text and sound effect. 

## Why It's Good For The Game
Adds a fun bit of noise and interaction if you have a riot shield and a combat knife.

## Images of changes
image

## Testing
1. Spawned in a riot shield, baton, combat knife
2. Checked that both items could bash the riot shield properly

## Changelog
:cl:
tweak: You can now bash riot shields with a combat knife to make a sound.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/shields.dm [hunk 1]
```diff
@@ -30,7 +30,7 @@
 	var/cooldown = 0 //shield bash cooldown. based on world.time
```
Comment (Contrabang) [inline code/game/objects/items/weapons/shields.dm:31]:
```suggestion
var/list/allowed_bashers = list(/obj/item/melee/baton, /obj/item/kitchen/knife/combat)
```

File: code/game/objects/items/weapons/shields.dm [hunk 2]
```diff
@@ -30,7 +30,7 @@
...
 
 /obj/item/shield/riot/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/melee/baton))
+	if(istype(W, /obj/item/melee/baton) || istype(W, /obj/item/kitchen/knife/combat))
```
Comment (Contrabang) [inline code/game/objects/items/weapons/shields.dm:33]:
```suggestion
	if(is_type_in_list(W, allowed_bashers))
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
Can you make this use checking in a list? Either like this, or with typecaches

**Review by Contrabang — APPROVED** [review]
LGTM

**Review by DGamerL — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**